src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joined = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if ((this[i] === null || this[i] === undefined)
+        && i === this.length - 1) {
+        continue;
+      } else if (this[i] === null || this[i] === undefined) {
+        joined += separator;
+      } else if (i === this.length - 1) {
+        joined += this[i];
+      } else {
+        joined = joined + this[i] + separator;
+      }
```

Comment:
can be simplified to these 2 actions
first - check if this[i] has value and add it
second - add separator excluding last index